gateway-service/src/test/java/org/zowe/apiml/gateway/acceptance/xForwardHeaders/MutateRemoteAddressFilter.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml.gateway.acceptance.xForwardHeaders;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+import org.springframework.core.Ordered;
+import org.springframework.core.annotation.Order;
+import org.springframework.web.server.ServerWebExchange;
+import org.springframework.web.server.WebFilter;
+
+import java.net.InetSocketAddress;
+import java.util.concurrent.atomic.AtomicReference;
+
+@Configuration
+@Profile("forward-headers-proxy-test")
+public class MutateRemoteAddressFilter {
+
+    public AtomicReference<String> proxyAddressReference;
+    private String originalProxyAddressProperty;
```

Comment:
Still same question as in V2, which looks like you didn't apply. Could it be `final`?
```suggestion
    private final String originalProxyAddressProperty;
```

Reply:
weird, I thought I did